public/components/trace_analytics/components/traces/span_detail_panel.tsx

```diff
@@ -135,72 +135,98 @@ export function SpanDetailPanel(props: {
     }
   };
 
-  const refresh = debounce(() => {
-    if (isEmpty(props.colorMap)) return;
-    const refreshDSL = spanFiltersToDSL();
-    setDSL(refreshDSL);
-    handleSpansGanttRequest(
-      props.traceId,
-      props.http,
-      setData,
-      props.colorMap,
-      refreshDSL,
-      mode,
-      props.dataSourceMDSId
-    ).finally(() => setIsGanttChartLoading(false));
-  }, 150);
-
-  const spanFiltersToDSL = () => {
-    const spanDSL: any =
-      mode === 'jaeger'
-        ? {
-            query: {
-              bool: {
-                must: [
-                  {
-                    term: {
-                      traceID: props.traceId,
-                    },
-                  },
-                ],
-                filter: [],
-                should: [],
-                must_not: [],
-              },
-            },
+  const parseAndFilterHits = (
+    payloadData: string,
+    traceMode: string,
+    payloadSpanFilters: any[]
+  ) => {
+    try {
+      const parsed = JSON.parse(payloadData);
+      let hits: any[] = [];
+      if (parsed.hits && Array.isArray(parsed.hits.hits)) {
+        hits = parsed.hits.hits;
+      } else if (Array.isArray(parsed)) {
+        hits = parsed;
+      } else {
+        console.warn('Unexpected payload format:', parsed);
+        return [];
+      }
+
+      hits = hits.map((hit) => {
+        if (traceMode === 'jaeger') {
+          if (!hit.sort || !hit.sort[0]) {
+            return {
+              ...hit,
+              sort: [Number(hit._source.startTime) * 1000], // Jaeger: startTime is in microseconds
+            };
           }
-        : {
-            query: {
-              bool: {
-                must: [
-                  {
-                    term: {
-                      traceId: props.traceId,
-                    },
-                  },
-                ],
-                filter: [],
-                should: [],
-                must_not: [],
-              },
-            },
-          };
-    spanFilters.map(({ field, value }) => {
-      if (value != null) {
-        spanDSL.query.bool.filter.push({
-          term: {
-            [field]: value,
-          },
+        } else {
+          if (!hit.sort || !hit.sort[0]) {
+            return {
+              ...hit,
+              sort: [parseIsoToNano(hit._source.startTime)],
+            };
+          }
+        }
+        return hit;
+      });
+
+      hits.sort((a, b) => b.sort[0] - a.sort[0]);
```

Comment:
do we use `hit.sort` anywhere else? if not then probably just do `hits.sort((a, b) => b._source.startTime - a._source.startTime);` here and remove the inserting sort logic above

Reply:
I believe the inserting sort logic is needed for handling jaeger vrs data prepper. Using just the hits.sort you show fails to render the gantt visualization.